interfaces/builtin/x11.go

```diff
@@ -60,14 +137,53 @@ socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
 bind
 `
 
+type x11Interface struct {
+	commonInterface
+}
+
+func (iface *x11Interface) AppArmorConnectedSlot(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	if !release.OnClassic {
+		old := "###PLUG_SECURITY_TAGS###"
+		new := plugAppLabelExpr(plug)
+		snippet := strings.Replace(x11ConnectedSlotAppArmor, old, new, -1)
+		spec.AddSnippet(snippet)
+	}
+	return nil
+}
+
+func (iface *x11Interface) SecCompPermanentSlot(spec *seccomp.Specification, slot *snap.SlotInfo) error {
+	if !release.OnClassic {
+		spec.AddSnippet(x11PermanentSlotSecComp)
+	}
+	return nil
+}
+
+func (iface *x11Interface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
+	if !release.OnClassic {
+		spec.AddSnippet(x11PermanentSlotAppArmor)
+	}
+	return nil
+}
+
+func (iface *x11Interface) UDevPermanentSlot(spec *udev.Specification, slot *snap.SlotInfo) error {
+	if !release.OnClassic {
+		spec.TagDevice(`KERNEL=="tty[0-9]*"`)
+		spec.TagDevice(`KERNEL=="mice"`)
+		spec.TagDevice(`KERNEL=="mouse[0-9]*"`)
+		spec.TagDevice(`KERNEL=="event[0-9]*"`)
+		spec.TagDevice(`KERNEL=="ts[0-9]*"`)
+	}
+	return nil
+}
+
 func init() {
-	registerIface(&commonInterface{
-		name:                  "x11",
-		summary:               x11Summary,
-		implicitOnClassic:     true,
-		baseDeclarationSlots:  x11BaseDeclarationSlots,
-		connectedPlugAppArmor: x11ConnectedPlugAppArmor,
-		connectedPlugSecComp:  x11ConnectedPlugSecComp,
-		reservedForOS:         true,
-	})
+	registerIface(&x11Interface{commonInterface{
+		name:                   "x11",
+		summary:                x11Summary,
+		implicitOnClassic:      true,
+		baseDeclarationSlots:   x11BaseDeclarationSlots,
+		connectedPlugAppArmor:  x11ConnectedPlugAppArmor,
+		connectedPlugSecComp:   x11ConnectedPlugSecComp,
+		rejectAutoConnectPairs: false,
```

Comment:
I suspect you specified this because it was part of commonInterface, but no other interfaces specify this. Perhaps drop it?

@zyga - please comment. What is rejectAutoConnectPairs meant to address? https://github.com/snapcore/snapd/pull/2113 first added it in 3bb44b8d7ec26ff69ad750fc67c14b85b664dad3 for the i2c interface, but i2c doesn't use it any more. This seems like dead code.

Reply:
I think I wanted the commonInterface AutoConnect to return true, and reading 

```
func (iface *commonInterface) AutoConnect(*interfaces.Plug, *interfaces.Slot) bool {
        return !iface.rejectAutoConnectPairs
}

```
and so I thought it was sensible. Happy to be told otherwise!

Reply:
I'll look into this as a part of my review pass, sorry for keeping you on hold.

Reply:
bump

Reply:
Note that rejectAutoConnectPairs defaults to false, you can safely drop it. False is the zero value for booleans.